services/job_runner.go

```diff
@@ -11,6 +12,116 @@ import (
 	"github.com/smartcontractkit/chainlink/utils"
 )
 
+// JobRunner safely handles coordinating job runs.
+type JobRunner interface {
+	Start() error
+	Stop()
+	ResumeSleepingRuns() error
+	ChannelForRun(string) chan<- store.RunRequest
+	WorkerCount() int
+}
+
+type jobRunner struct {
+	closer      chan struct{}
+	store       *store.Store
+	workerMutex sync.Mutex
+	workers     map[string]chan store.RunRequest
+}
+
+// NewJobRunner initializes a JobRunner.
+func NewJobRunner(str *store.Store) JobRunner {
+	return &jobRunner{
+		closer:  make(chan struct{}),
+		store:   str,
+		workers: make(map[string]chan store.RunRequest),
+	}
+}
+
+// Start reinitializes runs and starts the execution of the store's runs.
+func (rm *jobRunner) Start() error {
+	go rm.executeRuns()
+	return rm.ResumeSleepingRuns()
+}
+
+// ResumeSleepingRuns enqueues all recorded sleeping runs to make sure they are
+// woken after their sleep is expected to finish.
+func (rm *jobRunner) ResumeSleepingRuns() error {
+	pendingRuns, err := rm.store.JobRunsWithStatus(models.RunStatusPendingSleep)
+	if err != nil {
+		return err
+	}
+	for _, run := range pendingRuns {
+		rm.store.RunChannel.Send(models.RunResult{JobRunID: run.ID}, nil)
+	}
+	return nil
+}
+
+func (rm *jobRunner) executeRuns() {
```

Comment:
The method name `executeRuns` is too vague for my personal taste. I would call it `demultiplexRuns` to convey that we're delegating the RunResult to its respective `ChannelForRun`.